h/static/scripts/reducers/annotations.js

```diff
@@ -258,6 +258,13 @@ function savedAnnotations(state) {
   });
 }
 
+/** Return true if the annotation with a given ID is currently loaded. */
+function annotationExists(state, id) {
+  return state.annotations.some(function (annot) {
+    return annot.id === id;
```

Comment:
The number of places where we're having to do an `O(N)` operation in on `state.annotations` is beginning to concern me. Especially in situations like this where it would be perfectly reasonable to assume that `annotationExists` is a cheap operation.

At some point soon, let's convert the internal state into a map, or a pair of maps for `id`/`$$tag`.